avm/res/app/session-pool/main.bicep

```
@@ -0,0 +1,273 @@
+metadata name = 'Container App Session Pool'
+metadata description = 'This module deploys a Container App Session Pool.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Container App Session Pool.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Required. The container type of the sessions.')
+@allowed(['PythonLTS', 'CustomContainer'])
+param containerType string
+
+@description('Optional. Custom container definitions. Only required if containerType is CustomContainer.')
+param containers sessionContainerType
+
+@description('Optional. Required if containerType == \'CustomContainer\'. Target port in containers for traffic from ingress. Only required if containerType is CustomContainer.')
+param targetIngressPort int?
+
+@description('Optional. Container registry credentials. Only required if containerType is CustomContainer and the container registry requires authentication.')
+param registryCredentials sessionRegistryCredentialsType?
+
+@description('Optional. The cooldown period of a session in seconds.')
+param cooldownPeriodInSeconds int = 300
+
+@description('Optional. The maximum count of sessions at the same time.')
+param maxConcurrentSessions int = 5
+
+@description('Optional. The minimum count of ready session instances.')
+param readySessionInstances int?
+
+@description('Optional. Network status for the sessions. Defaults to EgressDisabled.')
+@allowed(['EgressEnabled', 'EgressDisabled'])
+param sessionNetworkStatus string = 'EgressDisabled'
+
+@description('Optional. The pool management type of the session pool. Defaults to Dynamic.')
+@allowed(['Dynamic', 'Manual'])
+param poolManagementType string = 'Dynamic'
+
+import { lockType } from 'br/public:avm/utl/types/avm-common-types:0.2.1'
+@description('Optional. The lock settings of the service.')
+param lock lockType?
+
+import { roleAssignmentType } from 'br/public:avm/utl/types/avm-common-types:0.2.1'
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType[]?
+
+import { managedIdentityAllType } from 'br/public:avm/utl/types/avm-common-types:0.2.1'
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentityAllType?
+
+@description('Optional. Settings for a Managed Identity that is assigned to the Session pool.')
+param managedIdentitySettings managedIdentitySettingType?
```

Udt spec 2/2
Types should be nonnullable and singular. Ref spec https://azure.github.io/Azure-Verified-Modules/specs/bcp/res/

To comply to that, you can add `[]?` here and remove it from the type definition
```suggestion
param managedIdentitySettings managedIdentitySettingType[]?
```